### ISSUE TITLE
[bug]: Active and focus state on header elements are broken

### ISSUE THREAD

**Issue by leagrdv** [swisspost/design-system#6558]
### 🐞 Describe the bug

Some styles are not correct anymore in the header, namely the "active + focused" state.
The focus ring should be outside of the element but on the language switch and on megadropdown links, it is not the case anymore, making it difficult to distinguish active and "active + focused" state.

#### ✨ Expected behavior

<img width="73" height="57" alt="Image" src="https://github.com/user-attachments/assets/f17faf46-c549-4b15-8701-8520f9b409ba" />

#### 💥 Current behavior

<img width="176" height="80" alt="Image" src="https://github.com/user-attachments/assets/99093c76-6a25-45e2-be3c-19603699713e" />

#### ✨ Expected behavior

<img width="181" height="56" alt="Image" src="https://github.com/user-attachments/assets/8a86b23e-300e-40f5-8135-79ea5a18f4d6" />

#### 💥 Current behavior

<img width="926" height="181" alt="Image" src="https://github.com/user-attachments/assets/179f9664-aeff-422e-a83f-036c687e441b" />
